front/src/modules/trainschedule/components/Timetable/Timetable.tsx

```diff
@@ -246,65 +359,124 @@ export default function Timetable({
+            >
+              <BiSelectMultiple />
+            </button>
+          )}
+          {multiselectOn && (
+            <>
+              <button
+                disabled={!selectedTrainIds.length}
+                className={cx('multiselect-delete', { disabled: !selectedTrainIds.length })}
+                type="button"
+                onClick={() =>
+                  openModal(
+                    <DeleteModal
+                      handleDelete={handleTrainsDelete}
+                      items={t('common/itemTypes:trains', { count: selectedTrainIds.length })}
+                    />,
+                    'sm'
+                  )
+                }
+              >
+                <GoTrash />
+              </button>
+              <input
+                type="checkbox"
+                className="mr-2"
+                checked={selectedTrainIds.length === trainsList.length}
+                onChange={() => selectAllTrains()}
+              />
+            </>
+          )}
+        </div>
+        <div className="d-flex">
           {multiselectOn && <span>{selectedTrainIds.length} / </span>}
           {t('trainCount', {
             count: trainsList.length,
           })}
         </div>
-        <div className="flex-grow-1">
-          <InputSNCF
-            type="text"
-            id="scenarios-filter"
-            name="scenarios-filter"
-            value={filter}
-            onChange={(e) => setFilter(e.target.value)}
-            placeholder={t('filterPlaceholder')}
-            whiteBG
-            noMargin
-            unit={<i className="icons-search" />}
-            sm
-            data-testid="scenarios-filter"
-          />
-        </div>
-        {!isEmpty(trainsList) && (
-          <button
-            type="button"
-            className={cx('multiselect-toggle', { on: multiselectOn })}
-            onClick={() => setMultiselectOn(!multiselectOn)}
-          >
-            <BiSelectMultiple />
-          </button>
-        )}
 
-        {multiselectOn && (
-          <button
-            disabled={!selectedTrainIds.length}
-            className={cx('multiselect-delete', { disabled: !selectedTrainIds.length })}
-            type="button"
-            onClick={() =>
-              openModal(
-                <DeleteModal
-                  handleDelete={handleTrainsDelete}
-                  items={t('common/itemTypes:trains', { count: selectedTrainIds.length })}
-                />,
-                'sm'
-              )
-            }
-          >
-            <GoTrash />
-          </button>
-        )}
+        <button type="button" className="filter-toggle btn btn-sm btn-only-icon">
+          <GoFilter onClick={toggleFilterPanel} />
+        </button>
       </div>
+      {isFilterPanelOpen && (
+        <div className="filter-panel">
+          <div className="row">
+            <div className="col-5">
+              <InputSNCF
+                type="text"
+                id="scenarios-filter"
```

Comment:
```suggestion
                id="timetable-label-filter"
```
As you don't really filter a list of scenarios, maybe you can name it differently, like this ?
Same thing for the `name`, and the `data-testid` attributes.

Reply:
Done